cmd/sys-mgmt-executor/main.go

```diff
@@ -83,107 +82,85 @@ func main() {
-
-			if status == "Up" {
-				if strings.Contains(line, "Up") {
-					LoggingClient.Info(fmt.Sprintf("container for service %s started: %s", service, line))
-					return true
-				} else {
-					LoggingClient.Warn(fmt.Sprintf("container for service %s NOT started", service))
-					return false
-				}
-			} else if status == "Exited" {
-				if strings.Contains(line, "Exited") {
-					LoggingClient.Info(fmt.Sprintf("container for service %s stopped: %s", service, line))
-					return true
-				} else {
-					LoggingClient.Warn(fmt.Sprintf("container for service %s NOT stopped", service))
-					return false
-				}
+	dec := json.NewDecoder(strings.NewReader(string(cmdOut)))
+	type containerInfo struct {
+		State struct {
+			Running bool
+		}
+	}
+
+	c := []containerInfo{}
+	for {
+		err = dec.Decode(&c)
+		if err != nil {
+			return false, err
+		}
+		switch {
+		case len(c) < 1:
+			return false, fmt.Errorf("container %s not found", service)
+		case len(c) > 1:
+			return false, fmt.Errorf("multiple containers found with name %s", service)
+		default:
+			if c[0].State.Running {
+				LoggingClient.Info(fmt.Sprintf("service container %s is running", service))
+			} else {
+				LoggingClient.Info(fmt.Sprintf("service container %s is not running", service))
 			}
+			return c[0].State.Running == running, err
 		}
 	}
-	return false
 }
 
 func ExecuteDockerCommands(service string, operation string) error {
-
+	// don't run commands for unknown services - could be an attack
 	if agent.IsKnownServiceKey(service) {
-		err := runDockerCommands(service, operation)
-		LoggingClient.Error(fmt.Sprintf("service %s ran into error while running Docker command: %v", service, err))
-		return err
-	} else {
-		err := fmt.Errorf("the service %s is an unknown service for which request was made to run Docker command", service)
-		LoggingClient.Error(err.Error())
-		return err
+		return runDockerCommands(service, operation)
 	}
+	return fmt.Errorf("the service %s is an unknown service for which request was made to run Docker command", service)
 }
 
 func runDockerCommands(service string, operation string) error {
-
-	var (
-		err    error
-		cmdDir string
-	)
-
-	cmdName := "docker"
-
-	cmdArgs := []string{operation, service}
-
 	// Validate that a known operation was requested.
 	if operation == START || operation == STOP || operation == RESTART {
-
-		cmd := exec.Command(cmdName, cmdArgs...)
-		cmd.Dir = cmdDir
-
-		out, err := cmd.CombinedOutput()
+		// run the docker command
+		out, err := exec.Command("docker", operation, service).CombinedOutput()
 		if err != nil {
-			LoggingClient.Error(err.Error())
-			LoggingClient.Info("docker command failed: %s", string(out))
+			LoggingClient.Error(fmt.Sprintf("docker command failed: %s", string(out)))
+			return err
 		}
 
+		// check that the command actually resulted in the correct state for
+		// the container
+		var expectedStatus bool
 		switch operation {
 		case START:
+			fallthrough
```

Comment:
I looked up the keyword `fallthrough` ([in the language reference](https://github.com/golang/go/wiki/Switch)) as I had not seen it being used _elsewhere_ in the `edgex-go` codebase... But this looks right to me, plus I found two instances of `fallthrough` (already) being used (both in `...internal/export/distro/registrations.go`), and which I had not noticed before 👍